public/app/features/alerting/unified/components/AlertManagerPicker.tsx

```diff
@@ -9,43 +9,38 @@ import { AlertManagerDataSource, GRAFANA_RULES_SOURCE_NAME } from '../utils/data
 
 interface Props {
   disabled?: boolean;
-  /**
-   * If true, only show alertmanagers that are receiving alerts from Grafana
-   */
-  showOnlyReceivingGrafanaAlerts?: boolean;
 }
 
 function getAlertManagerLabel(alertManager: AlertManagerDataSource) {
   return alertManager.name === GRAFANA_RULES_SOURCE_NAME ? 'Grafana' : alertManager.name.slice(0, 37);
 }
 
-export const AlertManagerPicker = ({ disabled = false, showOnlyReceivingGrafanaAlerts }: Props) => {
+export const AlertManagerPicker = ({ disabled = false }: Props) => {
   const styles = useStyles2(getStyles);
   const { selectedAlertmanager, availableAlertManagers, setSelectedAlertmanager } = useAlertmanager();
 
   const options: Array<SelectableValue<string>> = useMemo(() => {
-    return availableAlertManagers
-      .filter(({ name, handleGrafanaManagedAlerts }) => {
-        const isReceivingGrafanaAlerts = name === GRAFANA_RULES_SOURCE_NAME || handleGrafanaManagedAlerts;
-        return showOnlyReceivingGrafanaAlerts ? isReceivingGrafanaAlerts : true;
-      })
-      .map((ds) => ({
-        label: getAlertManagerLabel(ds),
-        value: ds.name,
-        imgUrl: ds.imgUrl,
-        meta: ds.meta,
-      }));
-  }, [availableAlertManagers, showOnlyReceivingGrafanaAlerts]);
+    return availableAlertManagers.map((ds) => ({
+      label: getAlertManagerLabel(ds),
+      value: ds.name,
+      imgUrl: ds.imgUrl,
+      meta: ds.meta,
+    }));
+  }, [availableAlertManagers]);
+
+  const isDisabled = disabled || options.length === 1;
+  const label = isDisabled ? 'Alertmanager' : 'Choose Alertmanager';
+
+  // User may have selected an alertmanager elsewhere in the application that has then ended up being filtered out
+  // In this case, we default back to Grafana AM
+  const selectedValue = options.some((am) => am.value === selectedAlertmanager)
+    ? selectedAlertmanager
+    : GRAFANA_RULES_SOURCE_NAME;
```

Comment:
This was originally because we might have filtered out the AM elsewhere, and then end up with nothing selected. This isn't needed for this purpose, but this might still end up mitigating https://github.com/grafana/grafana/issues/83320?)

Reply:
:+1: Yes, true - I've reverted the change and we can come back to it separately/in a more considered manner